### DISCUSSION TITLE
I can't get the hover infomation with using rust_analyzer api.

### DISCUSSION THREAD

**Discussion by baiguoname** [rust-lang/rust-analyzer#15593] [Q&A] [answered]
I have the following codeblock:
```
let a = 1;
```
I try to hover the information of `let` out, suppose the `let`'s textrange is `0..3`, then I do this (su-do code):
`api.hover(source_file_id, TextRange(0, 3))`, but the returned result is not from `let`, but from it's parent block. What am I doing wrong with the method `hover`?

**Comment by Veykril** [comment, marked as answer]
You want to call `api.hover(source_file_id, TextRange(0, 0))` (or 1,1 or 2,2). If you pass a non-empty range r-a will do a so called ranged hover which fetches the nearest expression and returns a type hover for that expression instead.

**Comment by baiguoname** [comment]
Thanks you so so so much!
